static/js/components/SourceDesktop.jsx

```diff
@@ -484,7 +484,10 @@ const SourceDesktop = ({ source }) => {
               </Typography>
             </AccordionSummary>
             <AccordionDetails>
-              <ObjPageAnnotations annotations={source.annotations} />
+              <ObjPageAnnotations
+                annotations={source.annotations}
+                spectra={spectra}
```

Comment:
Why not just pass in spectrum annotations after they've been processed rather than the entire spectra state?

Reply:
The annotations require additional info like the time each spectrum was detected. That means you have to start passing around data from spectra to annotations before giving it to the component. 
It seemed cleaner to just give the component this one optional argument.

Reply:
Well, for consistency, I think we should do the pre-processing before we pass the data in here so that we have consistent props like `annotations={...} specAnnotations = {...}`